src/workflows-app/pages/SubmissionHistory.test.js

```diff
@@ -0,0 +1,672 @@
+
+    getUser.mockReturnValue({
+      email: 'abc@gmail.com',
+    });
+
+    await Apps.mockImplementation(() => {
+      return {
+        listAppsV2: jest.fn(mockListAppsFn),
+      };
+    });
+
+    await Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runSets: {
+            get: jest.fn(getRunSetsMethod),
+          },
+        },
+      };
+    });
+
+    // Act
+    await act(async () => {
+      render(
+        h(BaseSubmissionHistory, {
+          name: 'test-azure-ws-name',
+          namespace: 'test-azure-ws-namespace',
+          workspace: mockAzureWorkspace,
+        })
+      );
+    });
+
+    await waitFor(() => {
+      expect(screen.getByRole('table')).toBeInTheDocument();
+    });
+
+    const table = screen.getByRole('table');
+
+    // Assert
+    expect(table).toHaveAttribute('aria-colcount', '6');
+    expect(table).toHaveAttribute('aria-rowcount', '3');
+
+    const rows = within(table).queryAllByRole('row');
+    expect(rows.length).toBe(3);
+
+    // check data rows are rendered as expected
+    const cellsFromDataRow1 = within(rows[1]).queryAllByRole('cell');
+    within(cellsFromDataRow1[headerPosition.Status]).getByText('Canceled');
+
+    const cellsFromDataRow2 = within(rows[2]).queryAllByRole('cell');
+    within(cellsFromDataRow2[headerPosition.Status]).getByText('Canceling');
+  });
+
+  it('should sort columns properly', async () => {
+    const getRunSetsMethod = jest.fn(() => Promise.resolve(runSetData));
+    const mockListAppsFn = jest.fn(() => Promise.resolve(mockAppResponse));
+
+    getUser.mockReturnValue({
+      email: 'abc@gmail.com',
+    });
+
+    await Apps.mockImplementation(() => {
+      return {
+        listAppsV2: jest.fn(mockListAppsFn),
+      };
+    });
+
+    await Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runSets: {
+            get: jest.fn(getRunSetsMethod),
+          },
+        },
+      };
+    });
+
+    // Act
+    await act(async () => {
+      render(
+        h(BaseSubmissionHistory, {
+          name: 'test-azure-ws-name',
+          namespace: 'test-azure-ws-namespace',
+          workspace: mockAzureWorkspace,
+        })
+      );
+    });
+
+    await waitFor(() => {
+      expect(screen.getByRole('table')).toBeInTheDocument();
+    });
+
+    const table = screen.getByRole('table');
+
+    const rows = within(table).queryAllByRole('row');
+    expect(rows.length).toBe(3);
+
+    const headers = within(rows[0]).queryAllByRole('columnheader');
+    expect(headers.length).toBe(6);
+
+    const topRowCells = (column) => {
+      const topRowCells = within(rows[1]).queryAllByRole('cell');
+      return topRowCells[column];
+    };
+
+    // Click on "Date Submitted" column and check that the top column is correct for:
+    // * ascending order
+    await act(async () => {
+      await fireEvent.click(within(headers[headerPosition['Date Submitted']]).getByRole('button'));
```

Comment:
fireEvent should be replaced by userEvent and extracted from the `act`, and user should be instantiated at the top of tests with `userEvent.setup()` https://github.com/DataBiosphere/terra-ui/pull/4074#discussion_r1273761408

Reply:
Changing the `fireEvent` to `user.click` causes an error for the `should sort columns properly` test. Despite getting the expected results, it fails with the message ` TypeError: Expected container to be an Element, a Document or a DocumentFragment but got undefined.`. Did you see this in your ticket at all?